src/catalog/catalog.ts

```diff
@@ -5,13 +5,15 @@ import {
     ICatalog,
     Key,
     PID,
+    SKU,
     SpecificTypedEntity,
 } from './interfaces';
 
 export class Catalog implements ICatalog {
     private mapGeneric = new Map<PID, GenericTypedEntity>();
     private mapSpecific = new Map<Key, SpecificTypedEntity>();
     private pidToKeys = new Map<PID, Key[]>();
+    private skuToKey = new Map<SKU, Key>();
```

Comment:
I wonder if the initialization and build up of this map (maybe all maps) shouldn't be deferred until first call. It's not free in terms of start-up overhead and obviously has some memory costs too. 

Perhaps a premature optimization, but I'm just throwing it out there for consideration.

Reply:
I agree with this being a bit premature.